examples/class_increment_semantic_segmentation/lifelong_learning_bench/testalgorithms/erfnet/ERFNet/utils/loss.py

```diff
@@ -0,0 +1,59 @@
+import torch
+import torch.nn as nn
+
+class SegmentationLosses(object):
+    def __init__(self, weight=None, size_average=True, batch_average=True, ignore_index=255, cuda=False, gpu_ids=0): # ignore_index=255
+        self.ignore_index = ignore_index
+        self.weight = weight
+        self.size_average = size_average
+        self.batch_average = batch_average
+        self.cuda = cuda
+        self.gpu_ids = gpu_ids
+
+    def build_loss(self, mode='ce'):
+        """Choices: ['ce' or 'focal']"""
+        if mode == 'ce':
+            return self.CrossEntropyLoss
+        elif mode == 'focal':
+            return self.FocalLoss
+        else:
+            raise NotImplementedError
+
+    def CrossEntropyLoss(self, logit, target):        
+        criterion = nn.CrossEntropyLoss()
+        if self.cuda:
+            criterion = criterion.cuda(self.gpu_ids)
+
+        loss = criterion(logit, target.long())
+
+        return loss
+
+    def FocalLoss(self, logit, target, gamma=2, alpha=0.5):
+        n, c, h, w = logit.size()
+        criterion = nn.CrossEntropyLoss(weight=self.weight, ignore_index=self.ignore_index,
+                                        size_average=self.size_average)
+        if self.cuda:
+            criterion = criterion.cuda(self.gpu_ids)
+
+        logpt = -criterion(logit, target.long())
+        pt = torch.exp(logpt)
+        if alpha is not None:
+            logpt *= alpha
+        loss = -((1 - pt) ** gamma) * logpt
+
+        if self.batch_average:
+            loss /= n
+
+        return loss
+
+if __name__ == "__main__":
+    loss = SegmentationLosses(cuda=True)
+    a = torch.rand(1, 3, 7, 7).cuda()
```

Comment:
What is the number '1 3 7 7 'meaning? if these lines is used for debug, please remove them.